### ISSUE TITLE
Lexicon - Navigation menu focus order on mobile

### ISSUE THREAD

**Issue by catbusofthecanals** [jadu/pulsar#1576]
**Describe the bug**

There are several bugs with focus order within the navigation menus:

1. At smaller viewports (width under 992 px), when expanding the primary nav menu focus isn't set on the first item within the nav menu but continues on into the main content of the page. Users have to backtab back through the menu to access the expanded content.

2. At smaller viewports (width under 992 px), when opening the secondary nav menu from the primary nav menu, focus lands on the Close secondary nav menu button but instead of proceeding into that menu’s first list item (eg. ‘Components’ in the lexicon) it jumps back to the primary nav menu Jadu logo.

**To Reproduce**
1. Go to Pulsar Lexicon
2. Set viewport to width under 992 px
3. Expand menu by activating Menu button
4. Tab throughout page using keyboard, note focus doesn't enter the expanded menu but continues throughout page
5. Backtab to enter primary nav menu
6. Activate link with secondary nav menu
7. Tab forwards, note that focus is set on 'Close' button but doesn't continue into expanded menu

**Expected behavior**
When a button is activated to expand a menu element, users would expect the next tabbable element to be the first item in the menu.

**Screenshots**
![Image](https://github.com/user-attachments/assets/dbbdb26a-a423-4ae6-b317-a578f86791ff)
![Image](https://github.com/user-attachments/assets/37b05bfd-67ea-4b2a-bd35-3a934c26a35d)

**Additional context**
Related success criteria:
[2.4.3 Focus Order (A)](https://www.w3.org/WAI/WCAG22/Understanding/focus-order.html)

**Suggested fixes**

1. After expanding the primary nav menu focus should be placed on the first item of the nav menu, eg. the Jadu logo and should then proceed through the list items until re-reaching the ‘Close’ button

2. After expanding the secondary nav menu focus should be placed on the first item of that nav menu, (eg. ‘Components’ within the Lexicon secondary nav menu) and should then proceed through the list items until looping back to the ‘Close’ button. Focus should be contained to the expanded menu but escapable with the ‘esc’ key, where focus should be set back on the primary menu list item.
